api/errors.go

```diff
@@ -0,0 +1,41 @@
+//nolint:lll
+
+package api
+
+import "fmt"
+
+// You need to run "go generate ./..." after making changes to this file
+//
+//go:generate go run golang.org/x/tools/cmd/stringer@v0.5.0 -linecomment -type ErrorCode
+type ErrorCode int
+
+const (
+	// Error codes in the 4001-4999 range are the user's fault,
+	// and error codes 5001-5999 are the server's fault, mimicking HTTP
+
+	ErrAddressMalformed           ErrorCode = 4000 + iota // address malformed
+	ErrAccountNotFound                                    // account not found
+	ErrTreasurerNotFound                                  // treasurer account not found
+	ErrOrgNotFound                                        // organization not found
+	ErrNoMetadataURI                                      // metadata provided but no metadata URI found in transaction
+	ErrWrongMetadataFormat                                // wrong metadata format
+	ErrMetadataURINotMatchContent                         // metadata URI does not match metadata content
+	ErrMissingStatusParameter                             // missing status parameter or unknown
+	ErrMarshalingJSONFailed                               // marshaling JSON failed
+)
+
+const (
+	ErrVochainEmptyReply        ErrorCode = 5000 + iota // vochain returned an empty reply
+	ErrVochainReturnedErrorCode                         // vochain replied with error code
+	ErrCantDecodeOrgID                                  // cannot decode organizationID
+	ErrCantDecodeAccountID                              // cannot decode accountID
+	ErrCantDecodeElectionID                             // cannot decode electionID
+	ErrCantParsePageNumber                              // cannot parse page number
+	ErrCantFetchElectionList                            // cannot fetch election list
+	ErrCantFetchElectionID                              // cannot fetch electionID
+	ErrCantFetchTokenTransfers                          // cannot fetch token transfers
+)
+
+func (e ErrorCode) Error() string {
+	return fmt.Sprintf("\"error\": {\"code\": %d, \"message\": \"%s\"}", e, e.String())
```

Comment:
to be clear, I'm not saying that the Error method should return JSON. This kind of `{code: x, message: y}` encoding should be done by the REST HTTP layer, which presumably responds to all JSON requests with a JSON response. The error value can be that object with two fields rather than just one string.

Reply:
The Error method here should just return the message, which is what will be useful when logging errors in Go code, for example. We could always make it something like `code: message` if we want to include both.

Reply:
makes absolute sense